enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,501 @@
+
+
+#### Single-node Deployments or MicroShift
+
+> TODO/TBD
+
+
+### Implementation Details/Notes/Constraints
+
+> WIP/TBReviewed
+
+- The implementation in CCM should handle the case where the `service.beta.kubernetes.io/aws-load-balancer-managed-security-group` annotation is set to `true` but the service type is not `NLB` (`aws-load-balancer-type: nlb`). In this scenario, the CCM should likely log a warning mentioning the annotation is supported only on NLB.
+- The initial implementation will focus on creating a single Security Group per NLB.
+- The Security Group naming convention should be consistent and informative, including the cluster ID, namespace, and service name to aid in identification and management in the AWS console. (TODO: need review, perhaps we just create the same name as LB?)
+- Egress rule management in CCM needs careful consideration to avoid overly permissive rules. The initial implementation should restrict egress to the necessary ports and protocols for communication with the backend pods (traffic ports and health check ports) within the cluster's VPC.
+- Proper IAM permissions for the CCM's service account will be required to allow it to create, describe, and delete Security Groups in AWS. This needs to be documented as a prerequisite. (TODO review: maybe we don't need this as IAM permissions would be already granted for CLB [review it])
+
+### Risks and Mitigations
+
+> WIP/TBReviewed
+
+- **Increased complexity in CCM**: Adding security group management to CCM increases its complexity. Mitigation: Focus on a minimal and well-tested implementation, drawing inspiration from the existing CLB security group management logic in CCM.   
+- **Potential for inconsistencies with ALBC**: If users later decide to migrate to ALBC, there might be inconsistencies in how security groups are managed. Mitigation: Clearly document the limitations of this approach and the benefits of using ALBC for more advanced scenarios or a broader range of features.
+- **Maintenance burden**: Maintaining this feature in CCM might become challenging if the upstream cloud-provider-aws project evolves significantly in its load balancer management. Mitigation: Upstream the changes to benefit from community maintenance and align with the long-term strategy for load balancer controllers in OpenShift.
+- **IAM Permissions(?)**: Incorrectly configured IAM permissions for the CCM could lead to failures in creating or managing security groups. Mitigation: Provide clear documentation on the necessary IAM permissions and potentially include checks in the CCM to verify these permissions.
+
+
+### Drawbacks
+
+> WIP/TBReviewed
+
+- The short-term solution requires engineering effort to implement and stabilize the changes in CCM and other components.
+- The extent of changes in CCM may require significant engineering effort from Red Hat to maintain this functionality. To mitigate it, this feature proposes minimum changes without disrupting existing flow.
```

Comment:
Is it really the minimum flow? I suspect the managed security group option isn't strictly required?